app/controllers/catalog_controller.rb

```diff
@@ -41,6 +42,48 @@ def update_user_per_cat_page
     end
   end
 
+  def submit_form # rubocop:disable MethodLength, AbcSize
```

Comment:
Another, more fundamental, comment - this method duplicates *a lot* of code from other methods, which is not good. I'd really like it if we could simply call the `ApplicationController#update_cart` method in here to deal with updating the dates and then something like [your code](https://github.com/YaleSTC/reservations/pull/1221/files#diff-956e46b33fb5307990c4e1a4b5fd86ccR71) to handle quantity updates. This would require tweaking the structure of the form and potentially the `update_cart` method, but it would be *much* more sustainable.